### PR TITLE
Upgrade to version 5.2.6

### PR DESCRIPTION
**Pull request by yalh76**
Upgrade to version 5.0.7
Can't make LDAP work.
I have set a password to choose in manifest instead of the random admin password.

### CODE REVIEW

**Comment by yalh76** [comment]
!testme

**Comment by zamentur** [comment]
@ericgaspar did you test the software manually ?

**Comment by ericgaspar** [comment]
I am not familiar with this app. I have installed testing but don't know how to interact with it 🤔.
I think it is worth to ask Limesurvey users to test it...

File: conf/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  # Include SSOWAT user panel.
+  include conf.d/yunohost_panel.conf.inc;
+}
+
+location ~ ^__PATH__/(data|config|\.ht|db_structure\.xml|README) {
+       deny all;
+}
+
+more_set_headers "X-Frame-Options : ALLOWALL";
```
Comment (alexAubin) [inline conf/nginx.conf]: (I understand that this is copypasta of the old nginx.conf.j2, but I'm suspicious about why this needs to be on the global scope and not inside the location)

File: manifest.json [hunk 2]
```diff
@@ -2,55 +2,50 @@
...
+        "en": "Creation and distribution survey tool",
+        "fr": "Outil de création et diffusion de sondage"
+    },
+    "version": "5.0.7~ynh1",
+    "url": "https://www.limesurvey.org/",
+    "upstream": {
+        "license": "GPL-2.0+",
+        "website": "https://www.limesurvey.org",
+        "admindoc": "https://manual.limesurvey.org/LimeSurvey_Manual/fr",
+        "userdoc": "https://yunohost.org/apps",
```
Comment (alexAubin) [inline manifest.json]:
```suggestion
```

File: manifest.json [hunk 3]
```diff
@@ -60,17 +55,22 @@
...
             },
             {
                 "name": "is_public",
                 "type": "boolean",
-                "ask": {
-                    "en": "Will people without YunoHost account have LimeSurvey account to be able to create polls ?",
-                    "fr": "Est-ce que des personnes sans compte YunoHost aurront un compte LimeSurvey pour créer des questionnaires ?"
+                "help": {
+                    "en": "If enabled, LimeSurvey will be accessible by people who do not have an account. This can be changed later via the webadmin.",
+                    "fr": "Si cette case est cochée, LimeSurvey sera accessible aux personnes n’ayant pas de compte. Vous pourrez changer ceci plus tard via la webadmin."
```
Comment (alexAubin) [inline manifest.json]:
```suggestion
                    "en": "In private mode, only authorized YunoHost members can create poll, with the public mode, it's possible to create account to people with no YunoHost account. ",
                    "fr": "En mode privé, seuls les utilisateurs YunoHost autorisés peuvent créer un sondage. En mode public, il est possible de créer un sondage sans compte YunoHost."
```

File: doc/DISCLAIMER.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+## YunoHost specific features
+
+* In private mode, only authorized YunoHost members can create poll, with the public mode, it's possible to create account to people with no YunoHost account.
```
Comment (alexAubin) [inline doc/DISCLAIMER.md]:
```suggestion
```

File: scripts/install [hunk 5]
```diff
@@ -20,166 +20,189 @@ ynh_abort_if_errors
...
+#ynh_add_config --template="../conf/data.sql" --destination="$final_path/data.sql"
+
+#ynh_mysql_connect_as --user="$db_user" --password="$db_pwd" --database="$db_name" < $final_path/data.sql
+
+#ynh_secure_remove --file=$final_path/data.sql
+
+#if [ $is_public -eq 1 ]
+#then
+#	ynh_mysql_connect_as --user="$db_user" --password="$db_pwd" --database="$db_name" <<< "UPDATE `lime_plugin_settings` SET value='\"0\"' WHERE `id`=21;"
+#fi
```
Comment (alexAubin) [inline scripts/install]: Uuuuuh it looks like those stuff were important ?

**Comment by ericgaspar** [comment]
!testme

**Comment by yalh76** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by zamentur** [comment]
ABout LDAP, i remember the account need to be created in bdd to get it work

**Comment by Shnoulle** [comment]
> ABout LDAP, i remember the account need to be created in bdd to get it work

Or : activate the autocreate system.

(I have to create a YNH Auth plugin and give it to YNH community).

**Comment by alexAubin** [comment]
!testme

**Comment by ericgaspar** [comment]
!tastemetitus

**Comment by zamentur** [comment]
!tastemetitus

**Comment by zamentur** [comment]
!tastemetitus

**Comment by yalh76** [comment]
!testme

File: manifest.json [hunk 6]
```diff
@@ -2,56 +2,49 @@
     "name": "LimeSurvey",
     "id": "limesurvey",
     "packaging_format": 1,
-    "version": "3.17.1~ynh1",
     "description": {
-        "en": "LimeSurvey is used to create advanced poll.",
-        "fr": "LimeSurvey est un outil de création et diffusion de sondage en ligne."
+        "en": "Creation and distribution survey tool",
```
Comment (alexAubin) [inline manifest.json]:
```suggestion
        "en": "Create and distribute surveys",
```

File: scripts/backup [hunk 7]
```diff
@@ -1,53 +1,78 @@
...
 source /usr/share/yunohost/helpers
 
 #=================================================
 # MANAGE SCRIPT FAILURE
 #=================================================
 
+ynh_clean_setup () {
+	### Remove this function if there's nothing to clean before calling the remove script.
+	true
+}
```
Comment (alexAubin) [inline scripts/backup]:
```suggestion
```

File: scripts/install [hunk 8]
```diff
@@ -20,166 +20,193 @@ ynh_abort_if_errors
...
-#=================================================
-# SECURING FILES AND DIRECTORIES
+# Add nice themes
 #=================================================
-ynh_script_progression --message="Apply permissions..." --time --weight=1
-# Set permissions
-set_permissions
+
+#ynh_setup_source "$datadir/upload/templates/libreform" libreform
+#ynh_setup_source "$final_path/upload/templates/librepoll" librepoll
```
Comment (alexAubin) [inline scripts/install:164]:
```suggestion
```

(No need to keep commented stuff)

File: scripts/install [hunk 9]
```diff
@@ -20,166 +20,193 @@ ynh_abort_if_errors
...
 
 #=================================================
 # SETUP FAIL2BAN
 #=================================================
-# ynh_script_progression --message="Configuring fail2ban..." --time --weight=1
-#ynh_add_fail2ban_config "/var/log/nginx/${domain}-error.log" "PHP message: Leed: wrong login for .* client: <HOST>" 5
+#ynh_script_progression --message="Configuring fail2ban..." --weight=1
+
+# No message in logs and the apps contains already a 10 minutes ban policy
```
Comment (alexAubin) [inline scripts/install]:
```suggestion
```

(No need to keep commented stuff)

File: doc/DISCLAIMER.md [hunk 10]
```diff
@@ -0,0 +1,4 @@
+## YunoHost specific features
+
+* SSO and LDAP are configured.
+* Login secured by fail2ban
```
Comment (alexAubin) [inline doc/DISCLAIMER.md]: Is it ? The install script seems to have no fail2ban conf in the end. Comment mentions that limesurvey already contains some brute forcing security, which is probably cool, but not a "yunohost specific feature" like the DISCLAIMER.md says ?

File: scripts/install [hunk 11]
```diff
@@ -20,166 +20,193 @@ ynh_abort_if_errors
...
+
+if [ $is_public -eq 1 ]
+then
+    allowed_groups="visitors $admin"
+else
+    allowed_groups="$admin"
+fi
+ynh_permission_update --permission="main" --show_tile="false" --add="visitors"
+
+ynh_permission_create --permission="admin" --url="/admin" --allowed=$allowed_groups  --show_tile="true"
```
Comment (alexAubin) [inline scripts/install:184]:
```suggestion
ynh_permission_create --permission="admin" --url="/admin" --allowed="$allowed_groups"  --show_tile="true"
```

(I doubt this works without quotes if `allowed_groups` contains spaces)

File: scripts/restore [hunk 12]
```diff
@@ -6,89 +6,124 @@
...
+source /usr/share/yunohost/helpers
 
 #=================================================
 # MANAGE SCRIPT FAILURE
 #=================================================
 
+ynh_clean_setup () {
+	#### Remove this function if there's nothing to clean before calling the remove script.
+	true
+}
```
Comment (alexAubin) [inline scripts/restore]:
```suggestion
```

File: scripts/upgrade [hunk 13]
```diff
@@ -1,153 +1,173 @@
...
-ynh_keep upload
+if [ "$upgrade_type" == "UPGRADE_APP" ]
+then
+    ynh_script_progression --message="Upgrading source files..." --weight=1
 
-# Copie new files and restore config, plugins, upload and themes
-ynh_secure_remove $final_path
-ynh_setup_source "$final_path"	# Télécharge la source, décompresse et copie dans $final_path
+    # Download, check integrity, uncompress and patch the source from app.src
+    ynh_setup_source --dest_dir="$final_path" --keep="$final_path/plugins/* $final_path/application/config/config.php $final_path/upload"
```
Comment (alexAubin) [inline scripts/upgrade]: Does `--keep` really works with wildcard ? Why not just `$final_path/plugins/` ?

Comment (zamentur) [reply scripts/upgrade]: Cause we want new plugins...
```suggestion
    ynh_setup_source --dest_dir="$final_path" --keep="$(echo $final_path/plugins/*) $final_path/application/config/config.php $final_path/upload"
```

File: scripts/install (hunk 11 above)
Comment (zamentur) [reply scripts/install:184]: I quote here the documentation:
> Usage: ynh_permission_create --permission="permission" [--url="url"] [--additional_urls="second-url" [ "third-url" ]] [--auth_header=true|false] [--allowed=group1 [ group2 ]] [--label="label"] [--show_tile=true|false] [--protected=true|false]

But you may be right...

File: doc/DISCLAIMER.md (hunk 10 above)
Comment (zamentur) [reply doc/DISCLAIMER.md]: Indeed
```suggestion
```

File: doc/DISCLAIMER.md [hunk 14]
```diff
@@ -0,0 +1,4 @@
+## YunoHost specific features
+
+* SSO and LDAP are configured.
```
Comment (zamentur) [inline doc/DISCLAIMER.md]:
```suggestion
* LDAP are configured.
```
SSO is disabled for security reasons

File: scripts/install (hunk 9 above)
Comment (zamentur) [reply scripts/install]:
```suggestion
# No message in logs and the apps contains already a 10 minutes ban policy
```
I keep the comment to explain why it's not a problem to have no fail2ban conf. For once I take the time to write a comment :p

**Comment by zamentur** [comment]
About the fact that the private mode is not working. In fact the URL / should be public to allow people to answer to the survey.
The public/private mode apply to /admin (to allow or not external user to create survey).

I don't know what i should do with this.

**Comment by zamentur** [comment]
!testme

File: scripts/install (hunk 11 above)
Comment (zamentur) [reply scripts/install:184]: I checked and it works like i wrote it (and i don't know if it works with the double quotes)

**Comment by zamentur** [comment]
!testme

File: doc/DISCLAIMER.md [hunk 15]
```diff
@@ -0,0 +1,3 @@
+## YunoHost specific features
+
+* LDAP are configured.
```
Comment (alexAubin) [inline doc/DISCLAIMER.md]:
```suggestion
* The application is integrated with YunoHost accounts (LDAP database)
```

**Review by alexAubin — APPROVED** [review]
Super quick re-review

**Comment by yalh76** [comment]
!testme

**Comment by Jaxom99** [comment]
Not much of a maintainer, but from looking at the test result, it looks like the CI test setup is wrong : the _private install_ launches this command `Running: yunohost app install --force /app_folder -a "domain=sub.domain.tld&path=/&admin=package_checker&language=en&is_public=1&is_admin_public=true&password=null".`

Shouldn't it be `is_public=0 `?

As always, thank for your amazing work mates :smiley_cat:

**Comment by yalh76** [comment]
There was an issue with the CI, it should be solved for now, I relaunched the test

**Comment by Jaxom99** [comment]
No luck so far, still this private test failure. Did the CI config actually change ?

**Comment by yalh76** [comment]
Launching some package on my own CI to see if the results are the same

**Comment by yalh76** [comment]
!testme

**Comment by yalh76** [comment]
!testme

**Pull request merged**